declaration/middleware.py

```diff
@@ -0,0 +1,47 @@
+import re
+
+from django.conf import settings
+from django.core.urlresolvers import reverse_lazy
+from django.http import HttpResponseRedirect
+from django.template.defaultfilters import urlencode
+
+from .views import DECLARATION_COOKIE_KWARGS
+
+
+PROTECTED_URL_PATTERNS = settings.PROTECTED_URL_PATTERNS
+
+
+class DeclarationMiddleware(object):
```

Comment:
I would like it to be compatible although it is not essential now. Is `django.utils.deprecation.MiddlewareMixin` available in earlier versions of Django?